app.rb

```diff
@@ -2,29 +2,33 @@
 require 'bundler/setup'
 require_relative 'publisher.rb'
 require_relative 'subscriber.rb'
+require_relative 'q_it_errors'
+
+SERVICES = ['publish', 'subscribe']
 
 def run_service(options)
   service = options.delete_at(0)
-  
   case service
   when 'publish'
+    Publisher.validate(options)
     publisher = Publisher.new(options)
     publisher.start
   when 'subscribe'
+    Subscriber.validate(options)
     subscriber = Subscriber.new(options)
     subscriber.start
   else
-    puts "Invalid service name. Use 'publish' or 'subscribe'."
-    exit(1)
-  end
+    raise QItNoServiceError, "Invalid service name. Try 'publish' or 'subscribe'." if (service.nil? || SERVICES.include?(service.downcase))
```

Comment:
Use `SERVICES` here again instead of `Try 'publish' or 'subscribe'`. So something like this
`".. Try any one of #{SERVICES.join(','}."`